### PR TITLE
Add feature flag for fake marking results as sensitive

### LINKED ISSUE #854: Add non-production feature flag for marking half of results as sensitive
For the sensitive content flow we are designing in #791 we need a way to test sensitive results. For two reasons, I'd like to create a feature flag which marks 50% of results as sensitive:

- Folks testing this functionality should not have to see *actual* sensitive content if they do not want to.
- Because marking results as sensitive is a largely manual process, we do not actually have many results marked as sensitive.

By creating this feature flag we can mark non-sensitive results as sensitive in order to test these features. I propose when this flag is active, we manipulate the API results to set every other result to have a true value in their `mature` field. 

This flag will also need to set `mature=true` to API requests.

This flag should be `disabled` in production, and switchable everywhere else.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #854 by @zackkrida 

## Description

This PR adds a feature flag `fake_sensitive` to mark roughly half the results as mature. It uses a random (but deterministic) approach which allows results to be marked mature consistently and across Pinia stores and browsing sessions.

This PR makes no UI changes. It is limited in scope to just adding the feature flag and setting up its behaviour. UI changes are already necessitated by #791 so this PR does not put in effort that's destined to be repeated.

## Testing Instructions

0. Check out the PR and run the dev server.
0. Enable the flag 'fake_sensitive' from `/preferences`.
0. Search for somethine.
   - You will see the console logs printing `Fake mature <uuid4>` for ~50% of results.
   - This list should remain the same no matter how many times you refresh the page.
0. Pick any ID out of this list and see the single result page for that ID. 
   - You should get the `Fake mature <uuid4>` log again because it also works on the single media store.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
This approach works perfectly. Awesome. Do you think it's worth creating a little utility function for the mature detection so the logic doesn't need to be updated in two places? Something like:

```
const fakeMature = (item) => {
  if (prng(hash(item.id))() < 0.5) return
  item.mature = true
  log("Fake mature", item.id)
}
```

I don't think it's necessary but something to consider.

I also tried this with some blurring, which was a fun way to test:

CleanShot 2023-03-08 at 08 31 45

**Comment by dhruvkb** [comment]
Good point, I'll extract it to a separate utility 👍

File: frontend/src/types/media.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ export interface Media {
 
   tags: Tag[]
   fields_matched?: string[]
+
+  mature: boolean
```
Comment (obulat) [inline frontend/src/types/media.ts:48]: Wow, I didn't know we didn't have this in the type! Great catch! Is it always available, or should we mark it as optional?

**Review by obulat — APPROVED** [review]
Really nice! I have a question about `mature` type inline, but don't want to block the PR.

File: frontend/src/types/media.ts (hunk 1 above)
Comment (dhruvkb) [reply frontend/src/types/media.ts:48]: It's always available.

**Pull request merged**